Kernel/Thread.cpp

```diff
@@ -366,6 +367,14 @@ ShouldUnblockThread Thread::dispatch_signal(u8 signal)
         return ShouldUnblockThread::Yes;
     }
 
+    ProcessPagingScope paging_scope(m_process);
+	/*
+	 * The userspace registers should be stored at the top of the stack
+	 * We have to subtract 2 because the processor decrements the kernel
+	 * stack before pushing the args.
+	 */
+    RegisterDump* regs = (RegisterDump*)(kernel_stack_top()-sizeof(RegisterDump)-2);
```

Comment:
This will never be null, so we should use a reference:
```suggestion
    auto& regs = *(RegisterDump*)(kernel_stack_top() - sizeof(RegisterDump) - 2);
```